nano/lib/utility.hpp

```diff
@@ -158,8 +158,11 @@ class observer_set final
 	}
 	void notify (T... args)
 	{
-		nano::lock_guard<nano::mutex> lock (mutex);
-		for (auto & i : observers)
+		nano::unique_lock<nano::mutex> lock (mutex);
+		auto observers_copy = observers;
+		lock.unlock ();
+
+		for (auto & i : observers_copy)
 		{
 			i (args...);
 		}
```

Comment:
Maybe we could hide private members under `private:` section so that no one outside can access this container in thread unsafe way

Reply:
And it was meant to be a quick change! :-D

Reply:
I'll do that on another PR, it is a different type of change.